c2corg_api/models/user.py

```diff
@@ -86,14 +94,18 @@ def update_validation_nonce(self, purpose, days):
         particular actions like changing password or validating an email. It
         must have a short lifespan to avoid unused nonces to become a security
         risk."""
+        if purpose != Purpose.registration and not self.email_validated:
+            # An account must be validated before any other action is tried.
+            raise Exception('Account not validated yet')
```

Comment:
Yes, here is what is sent when a user registers, does not confirm the account, and then tries to request a new password:
```
HTTP/1.1 500 Internal Server Error
Content-Length: 110
Content-Type: text/plain
Date: Thu, 21 Apr 2016 07:19:39 GMT
Server: waitress

Internal Server Error

The server encountered an unexpected internal server error
```